oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+import os
+import re
+import shutil
+import time
+from pathlib import Path
+
+import pandas as pd
+from Bio.Blast.Applications import NcbiblastnCommandline, NcbimakeblastdbCommandline
+from joblib import Parallel, delayed, parallel_backend
+
+from ._filter_base import ProbeFilterBase
+from ._utils import _load_probes_info, _write_output, _write_removed_genes
+
+
+class ProbeFilterBlastn(ProbeFilterBase):
+    def __init__(
+        self,
+        n_jobs,
+        file_transcriptome_fasta,
+        dir_output,
+        dir_annotations,
+        word_size,
+        percent_identity,
+        coverage,
+        probe_length_min,
+        probe_length_max,
+        ligation_region=0,
```

Comment:
we should not call it ligation region but maybe something like "seed_region", which is a region where no perfect matches are allowed. This should be given as tuples with start / end coordinates wrt the probe. We should also allow multiple of those regions per probe (e.g. if the user wants to make sure that no perfect matches are at beginning and end of the probe)